game/src/main/scala/io/aigar/game/Virus.scala

```diff
@@ -0,0 +1,48 @@
+package io.aigar.game
+
+import com.github.jpbetz.subspace.Vector2
+import io.aigar.game.serializable.Position
+
+object Virus {
+  final val Quantity = 15
+  final val Mass = 100
+}
+
+class Virus(var position: Vector2) {
+
+  def update(grid: Grid, players: List[Player]): Unit = {
+    if(detectCollisions(players)) {
+      // TODO Add the virus consumption into the cell
+      respawn(grid, players)
+    }
+  }
+
+  def detectCollisions(players: List[Player]): Boolean ={
+    for(player <- players) {
+      for(cell <- player.cells) {
+        // TODO Change the 1.1 value to the constant
+        if(cell.contains(position) && cell.mass > Virus.Mass * 1.1){
+          return true
+        }
+      }
+    }
+    false
+  }
+
+  def respawn(grid: Grid, players: List[Player]): Unit = {
+    val newPosition = grid.randomPosition
+
+    // We always want 15 viruses. How to we manage it without doing a while true loop?
```

Comment:
I'll do it as you said : try n times and then spawn it anywhere, but it could spawn inside a big cell, hence be very disadvantageous for this player avoiding Viruses since the beginning. 

To fix it, I'll make it the same way as `Resources` : if it fails to spawn, I'll retry to on the next tick. Now I'm not sure it Viruses should inherit from the same as Cell or Resources... Looks like the 3 would need to respawn on the map not in other cells so...